src/Middleware/RequestTimeMiddleware.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace EightPoints\Bundle\GuzzleBundle\Middleware;
+
+use EightPoints\Bundle\GuzzleBundle\DataCollector\HttpDataCollector;
+use Psr\Http\Message\RequestInterface;
+use GuzzleHttp\TransferStats;
+
+class RequestTimeMiddleware
+{
+    /** @var HttpDataCollector */
+    private $dataCollector;
+
+    /**
+     * @param HttpDataCollector $dataCollector
+     */
+    public function __construct(HttpDataCollector $dataCollector)
+    {
+        $this->dataCollector = $dataCollector;
+    }
+
+    /**
+     * @param callable $handler
+     *
+     * @return \Closure
+     */
+    public function __invoke(callable $handler) : \Closure
```

Comment:
return callable maybe ?

Reply:
No, there Closure is created. You can see here: http://php.net/manual/en/functions.anonymous.php When you create anonymous function you create instance of Closure class.

Reply:
my bad